backend/ethereum/channel/conclude.go

```diff
@@ -119,6 +133,30 @@ func (a *Adjudicator) isConcluded(ctx context.Context, sub *subscription.EventSu
 	return false, errors.WithMessage(<-subErr, "reading past events")
 }
 
+// isForceExecuted returns whether a channel is already concluded.
+func (a *Adjudicator) isForceExecuted(ctx context.Context, c channel.ID) (bool, error) {
+	sub, err := subscription.NewEventSub(ctx, a.ContractBackend, a.bound, updateEventType(c), startBlockOffset)
+	if err != nil {
+		return false, errors.WithMessage(err, "subscribing")
+	}
+	defer sub.Close()
+	events := make(chan *subscription.Event, 10)
+	subErr := make(chan error, 1)
+	// Write the events into events.
+	go func() {
+		defer close(events)
+		subErr <- sub.ReadPast(ctx, events)
+	}()
+	// Read all events and check for force execution.
+	for _e := range events {
+		e := _e.Data.(*adjudicator.AdjudicatorChannelUpdate)
+		if e.Phase == phaseForceExec {
```

Comment:
Does it matter that you do not check all events? Can something after the force-exec happen?

Reply:
Currently, the only phase after FORCE_EXEC is CONCLUDED. Anyhow, it's a good observation. I will make sure that I check the phase of the last event, because this was the intention.